lib/xipio/middleware.rb

```diff
@@ -5,7 +5,7 @@ def initialize(app)
     end
 
     def call(env)
-      tld_length = env['HTTP_HOST'].end_with?('.xip.io') ? 6 : 1
+      tld_length = env['HTTP_HOST'].match(/\.xip\.io(:[0-9]{,5})?\z/) ? 6 : 1
```

Comment:
Could you change it to this?

```ruby
tld_length = env['HTTP_HOST'].match?(/\.xip\.io(:\d{1,5})?\z/) ? 6 : 1
```

`#match?` just returns a boolean, whereas `#match` returns the match data.